src/idom/server/flask.py

```diff
@@ -103,27 +103,53 @@ def run_server() -> None:  # pragma: no cover
         await stopped.wait()
     finally:
         # we may have exitted because this task was cancelled
-        server.stop(3)
+        server.current.shutdown()
         # the thread should eventually join
         thread.join(timeout=3)
         # just double check it happened
         if thread.is_alive():  # pragma: no cover
             raise RuntimeError("Failed to shutdown server.")
 
 
+def use_location() -> Location:
+    """Get the current route as a string"""
+    conn = use_connection()
+    search = conn.request.query_string.decode()
+    return Location(pathname="/" + conn.path, search="?" + search if search else "")
+
+
+def use_scope() -> dict[str, Any]:
+    """Get the current WSGI environment"""
+    return use_request().environ
```

Comment:
I named this `use_scope` because the WSGI environ is [very similar to the ASGI scope](https://asgi.readthedocs.io/en/latest/specs/www.html?highlight=environ#wsgi-compatibility) and I wanted a minimal common interface that all server implementations shared. Given that, using the term "scope" here instead of "environ" didn't seem too egregious.

Reply:
Do we not have "real WS scope" provided to us by flask-sock?

Reply:
Not that I can tell. All that's provided on the WS is an `environ` attribute. Under the hood it's using [this package](https://github.com/miguelgrinberg/simple-websocket) for WS support.